File: MAP_Lab7/MAP_Lab7/Commands/Command.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_Lab7.Commands
{
    public abstract class Command
    {
        public string key, description;

        public Command(string k, string d)
        {
            key = k;
            description = d;
        }

        public abstract void Execute();

        public string GetKey() { return key; }

        public string GetDescription() { return description; }
    }
}

File: Toy Interpreter/src/Model/Command/Command.java
/**************************************************************************************************
 *       Copyright(c):                                                                            *
 *                         Program made by @Moldovan Alexandru-Vasile.                            *
 *                                                                                                *
 **************************************************************************************************/

package Model.Command;

public abstract class Command
{
    private String key,description;

    public Command(String k, String descr)
    {
        key = k;
        description = descr;
    }

    public abstract void execute();

    public String getKey() {
        return key;
    }

    public String getDescription() {
        return description;
    }
}
